UnleashClient/constraints/Constraint.py

```diff
@@ -82,37 +82,42 @@ def check_string_operators(self, context_value: str) -> bool:
 
     def check_numeric_operators(self, context_value: Union[float, int]) -> bool:
         return_value = False
+
         parsed_value = float(self.value)
+        parsed_context = float(context_value)
 
         if self.operator == ConstraintOperators.NUM_EQ:
-            return_value = context_value == parsed_value
+            return_value = parsed_context == parsed_value
         elif self.operator == ConstraintOperators.NUM_GT:
-            return_value = context_value > parsed_value
+            return_value = parsed_context > parsed_value
         elif self.operator == ConstraintOperators.NUM_GTE:
-            return_value = context_value >= parsed_value
+            return_value = parsed_context >= parsed_value
         elif self.operator == ConstraintOperators.NUM_LT:
-            return_value = context_value < parsed_value
+            return_value = parsed_context < parsed_value
         elif self.operator == ConstraintOperators.NUM_LTE:
-            return_value = context_value <= parsed_value
-
+            return_value = parsed_context <= parsed_value
         return return_value
 
 
-    def check_date_operators(self, context_value: datetime) -> bool:
+    def check_date_operators(self, context_value: Union[datetime, str]) -> bool:
```

Comment:
I realise this is a very debatable change, however, all the other SDKs ingest a string at this point and that's also what our docs say. I've let a datetime continue to work here because there's a reasonable chance this is already in the wild

Reply:
Oh that's interesting!  I was assuming we'd want to favor language-specific date constructs. 😓

Reply:
Most of the client SDKs have grown up around the spec so they're a little more liberal because it can be a pain to guess the types in the context properties in client specs. I think the Python SDK made the right choices here, which is part of the reason I left this as a Union type